build failure on gcc 14

Hard to find a specific quote to support this, but it appears that implicit declarations became errors as of GCC 14 for `-std=c99`. Here's an example build failure as encountered in cctbx_project XFEL CI with GCC 14.3.0:
```
2025-07-21T16:20:43.6413610Z /__w/1/miniforge/envs/py311/bin/x86_64-conda-linux-gnu-cc -o lunus/c/src/lunus.o -c -std=c99 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /__w/1/miniforge/envs/py311/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /__w/1/miniforge/envs/py311/include -fopenmp -I/__w/1/modules/lunus/c/include /__w/1/modules/lunus/c/src/lunus.c
2025-07-21T16:20:43.6879526Z /__w/1/modules/lunus/c/src/lunus.c: In function 'main':
2025-07-21T16:20:43.6884934Z /__w/1/modules/lunus/c/src/lunus.c:246:28: error: implicit declaration of function 'getline' [-Wimplicit-function-declaration]
2025-07-21T16:20:43.6885962Z   246 |       while ((chars_read = getline(&json_name,&bufsize,f)) != -1) {
2025-07-21T16:20:43.6886768Z       |                            ^~~~~~~
```
where last week on GCC 13.3.0 we had:
```
2025-07-20T02:53:31.0299515Z /__w/1/miniforge/envs/py311/bin/x86_64-conda-linux-gnu-cc -o lunus/c/src/lunus.o -c -std=c99 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /__w/1/miniforge/envs/py311/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /__w/1/miniforge/envs/py311/include -fopenmp -I/__w/1/modules/lunus/c/include /__w/1/modules/lunus/c/src/lunus.c
2025-07-20T02:53:31.0675926Z /__w/1/modules/lunus/c/src/lunus.c: In function 'main':
2025-07-20T02:53:31.0680125Z /__w/1/modules/lunus/c/src/lunus.c:246:28: warning: implicit declaration of function 'getline' [-Wimplicit-function-declaration]
2025-07-20T02:53:31.0680777Z   246 |       while ((chars_read = getline(&json_name,&bufsize,f)) != -1) {
2025-07-20T02:53:31.0681234Z       |                            ^~~~~~~
```

--- a/c/include/lunus.h
+++ b/c/include/lunus.h
@@ -8,6 +8,11 @@
 			65535
 */
 
+/*
+ * Ensure that getline is declared when including stdio.h
+ */
+#define _POSIX_C_SOURCE 200809L
+
 #ifndef __MWMASK_H
 
 #define __MWMASK_H

